embulk-core/src/main/java/org/embulk/jruby/JRubyClassLoader.java

```diff
@@ -0,0 +1,120 @@
+package org.embulk.jruby;
+
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.Collection;
+import java.util.Enumeration;
+import java.util.Vector;
+
+/**
+ * Loads classes and resources of JRuby artifact(s).
+ *
+ * <p>It prioritizes a class found by this class loader over a class loaded by the parent class loader because
+ * this class loader is specific for JRuby. {@code jruby-complete-9.X.Y.Z.jar} contains all libraries required
+ * in JRuby. Some of them conflict with {@code embulk-core}'s top-level dependencies, such as {@code "asm:asm"}
+ * and {@code "org.sonatype.sisu.inject:cglib"} used from Guice.
+ *
+ * <p>JRuby should use JRuby's own dependencies, and Embulk should not be impacted from JRuby's depencencies.
```

Comment:
nit: typo in `depencencies`.

Reply:
Fixed in ef7af455559cec2eaf52ed5541bc503de7ae3cf0.